indicoio/utils/api.py

```
@@ -37,10 +63,33 @@ def api_handler(arg, cloud, api, url_params=None, **kwargs):
     headers = dict(JSON_HEADERS)
     headers["X-ApiKey"] = url_params.get("api_key") or config.api_key
     url = create_url(url_protocol, host, api, dict(kwargs, **url_params))
+    return collect_api_results(input_data, url, headers, batch_size, kwargs)
+
 
+def collect_api_results(input_data, url, headers, batch_size, kwargs):
+    """
+    Optionally split up a single request into a series of requests
+    to ensure timely HTTP responses.
+
+    Could eventually speed up the time required to receive a response by
+    sending batches to the indico API concurrently
+    """
+    if batch_size:
+        results = []
+        for batch in batched(input_data, size=batch_size):
+            results.extend(send_request(batch, url, headers, kwargs))
```

How are we thinking about handling errors? Let them be raised and exit out of batched or localize errors to each "batch" and include the IndicoError in the result list..